tests/test_multipart_empty_final.py

```diff
@@ -0,0 +1,60 @@
+"""Test that multipart reader handles empty parts at the end correctly."""
+
+import pytest
+
+from aiohttp import web
+from aiohttp.multipart import BodyPartReader, MultipartReader
+from aiohttp.test_utils import AioHTTPTestCase
+
+
+class TestMultipartEmptyFinal(AioHTTPTestCase):
+    async def get_application(self) -> web.Application:
+        app = web.Application()
+        app.router.add_post("/graphql", self.multipart_handler)
+        return app
+
+    async def multipart_handler(self, request: web.Request) -> web.StreamResponse:
+        response = web.StreamResponse()
+        response.headers["Content-Type"] = "multipart/mixed; boundary=graphql"
+        await response.prepare(request)
+
+        # preamble
+        await response.write(
+            b"Content-Type: application/json\r\n"
+            b"Content-Length: 2\r\n"
+            b"\r\n"
+            b"{}\r\n"
+        )
+
+        # valid part
+        await response.write(
+            b"--graphql\r\n"
+            b"Content-Type: application/json\r\n"
+            b"Content-Length: 15\r\n"
+            b"\r\n"
+            b'{"data": "foo"}\r\n'
+        )
+
+        # empty part which triggers the bug - technically valid
+        await response.write(b"--graphql\r\n")
+
+        await response.write(b"--graphql--")
```

Comment:
```suggestion
        await response.write(b"\r\n--graphql--")
```
Just to sanity check that it does work with syntactically valid empty body parts.